MapboxNavigation/RouteMapViewController.swift

```diff
@@ -253,6 +259,66 @@ extension RouteMapViewController: NavigationMapViewDelegate {
         let route = routeController.routeProgress.route
         guard let coordinates = route.coordinates else  { return nil }
         
+        if let userLocation = mapView.userLocation,
+            let style = mapView.style,
+            recenterButton.isHidden {
+            
+            let streetsLanguages = ["zh", "ru", "fr", "es", "en", "de"]
+            let roadLabelLayerIdentifier = "roadLabelLayer"
+            
+            let streetsSources = style.sources.flatMap {
+                $0 as? MGLVectorSource
+                }.filter {
+                    $0.isMapboxStreets
+            }
+            let streetsSourceIdentifiers = streetsSources.map {
+                $0.identifier
+            }
+            assert(!streetsSourceIdentifiers.isEmpty, "Style must contain the source `mapbox.mapbox-streets-v7`")
+            
+            if let mapboxSteetsSource = streetsSources.first, style.layer(withIdentifier: roadLabelLayerIdentifier) == nil{
+                let streetLabelLayer = MGLLineStyleLayer(identifier: roadLabelLayerIdentifier, source: mapboxSteetsSource)
+                streetLabelLayer.sourceLayerIdentifier = "road_label"
+                
+                // If the opacity is set to 0, the feature will be ignored in `mapView.visibleFeatures()`
+                streetLabelLayer.lineOpacity = MGLStyleValue(rawValue: 0.001)
+                streetLabelLayer.lineWidth = MGLStyleValue(rawValue: 10)
+                streetLabelLayer.lineColor = MGLStyleValue(rawValue: .white)
+                style.addLayer(streetLabelLayer)
+            }
+            
+            if let userPuck = mapView.view(for: userLocation),
+                let routeWayNames = routeController.routeProgress.currentLegProgress.currentStep.names {
+                let features = mapView.visibleFeatures(in: userPuck.frame, styleLayerIdentifiers: Set([roadLabelLayerIdentifier]))
+                
+                for feature in features {
+                    
+                    var key = "name"
+                    if let languages = Locale.preferredLanguages.first,
+                        let language = languages.components(separatedBy: "-").first,
+                        streetsLanguages.contains(language) || languages == "zh-Hans" {
+                        key += "_\(language)"
+                    }
+                    
+                    if let name = feature.attribute(forKey: key) as? String {
+                        
+                        let filteredStreetNames = routeWayNames.filter {
+                            return name.distanceFrom(string: $0) <= 6
+                        }
+                        
+                        if !filteredStreetNames.isEmpty {
```

Comment:
OK, I misread the code: a “filtered” street name must be w/i 6 of _any_ of the streets along the route. Presumably there would be an edit distance of 0 between Broadway Avenue and the Broadway Avenue in the route data. However, there’s also the problem of false positives: if the user drives along State Route 185 where it passes over State Route 85, the label will happily switch to “State Route 85” because of an edit distance of one.